### DISCUSSION TITLE
Inactive parts cannot be allocated to a build order. Add a setting?

### DISCUSSION THREAD

**Discussion by SergeoLacruz** [inventree/InvenTree#5285] [Ideas]
Hi,
An inactive part cannot be allocated to a build order. However  there might still be parts on stock that can be used up. I cannot find a setting that allows to allocate inactive items. Do I miss something? 
If not I think is is a nice to add it. 

Michael

**Comment by SchrodingersGat** [comment]
This is by intention - inactive parts are meant to be hidden from active use. If you need to use them, temporarily mark the part as "active"

**Reply by xeno27** [reply]
But I think there is a difference between obsolete parts and inactive parts. 
In the documents, you suggest using the inactive flag for obsolete parts, but as I see it, we can still use the obsolete parts that are on stock for old projects,  which then include using them for a build order. And do not use them for new projects.

**Reply by SergeoLacruz** [reply]
Exactly. We do not trash obsolete parts but use them until the stock is empty. Then we replace them by a new part. 
Ideally there are several value for the status like e.g. new part, in use, not recommended..., obsolete. But this usually work for a life cycle management system. 
Anyhow, there is already a setting that allows to list inactive parts in the search. I can imagine that this setting also allows to use the parts in builds. An additional setting was also fine. 
If we set them temporarily to active, the one who makes the allocations also needs to have write access to parts which is usually not the case.

**Reply by SchrodingersGat** [reply]
Currently, we do no support such a distinction

**Comment by matmair** [comment]
Adding more complex states could be achieved by #4969 but with the differing use cases for InvenTree, I am starting to think that a general rule engine would be a better solution for requirements like lifecycles, phased sunsetting and similar things.
